src/settings/toml/tests/deploy_target.rs

```diff
@@ -335,3 +335,540 @@ fn it_errors_on_deploy_target_routes_and_workers_dev_true() {
+
+    assert!(actual_deploy_target.is_err());
+
+    // when zone id is missing, error
+    let patterns = ["hostname.tld/*"];
+    let env_config = EnvConfig::zoned_multi_route("", patterns.to_vec());
+
+    let script = "top_level_zoneless_env_zoned_multi_route_no_zone_id";
+    let test_toml = WranglerToml::webpack_zoneless_with_env(script, true, env_name, env_config);
+    let toml_string = toml::to_string(&test_toml).unwrap();
+    let manifest = Manifest::new_from_string(toml_string).unwrap();
+
+    let actual_deploy_target = manifest.deploy_target(Some(env_name));
+
+    assert!(actual_deploy_target.is_err());
+
+    // when zone id is present, all good
+    let patterns = ["hostname.tld/*"];
+    let env_config = EnvConfig::zoned_multi_route(zone_id, patterns.to_vec());
+
+    let script = "top_level_zoneless_env_zoned_multi_route_no_zone_id";
+    let test_toml = WranglerToml::webpack_zoneless_with_env(script, true, env_name, env_config);
+    let toml_string = toml::to_string(&test_toml).unwrap();
+    let manifest = Manifest::new_from_string(toml_string).unwrap();
+
+    let actual_deploy_target = manifest.deploy_target(Some(env_name)).unwrap();
+
+    let expected_name = manifest.worker_name(Some(env_name));
+
+    let expected_routes = patterns
+        .iter()
+        .map(|p| Route {
+            script: Some(expected_name.to_string()),
+            pattern: p.to_string(),
+            id: None,
+        })
+        .collect();
+
+    let expected_deploy_target = DeployTarget::Zoned(Zoned {
+        zone_id: zone_id.to_string(),
+        routes: expected_routes,
+    });
+
+    assert_eq!(actual_deploy_target, expected_deploy_target);
+}
+
+#[test]
+fn when_top_level_zoned_env_empty() {
+    let zone_id = "samplezoneid";
+    let pattern = "hostname.tld/*";
+    let env_name = "test";
+
+    let env_config = EnvConfig::default();
+
+    let script = "top_level_zoned_env_empty";
+    let test_toml = WranglerToml::webpack_zoned_single_route_with_env(
+        script, zone_id, pattern, env_name, env_config,
+    );
+    let toml_string = toml::to_string(&test_toml).unwrap();
+    let manifest = Manifest::new_from_string(toml_string).unwrap();
+
+    let actual_deploy_target = manifest.deploy_target(Some(env_name)).unwrap();
+
+    let expected_name = manifest.worker_name(Some(env_name));
+
+    let expected_routes = vec![Route {
+        script: Some(expected_name),
+        pattern: pattern.to_string(),
+        id: None,
+    }];
+    let expected_deploy_target = DeployTarget::Zoned(Zoned {
+        zone_id: zone_id.to_string(),
+        routes: expected_routes,
+    });
+
+    assert_eq!(actual_deploy_target, expected_deploy_target);
+}
+
+#[test]
+fn when_top_level_zoned_env_zoneless() {
+    let zone_id = "samplezoneid";
+    let pattern = "hostname.tld/*";
+    let env_name = "test";
+
+    // TODO: SHOULD THIS TEST PASS?
+    // when env.workers_dev = false
+    let env_config = EnvConfig::zoneless(false);
```

Comment:
this should pass and inherit the top level zoned deploy configs, right?

Reply:
i discussed this with avery; if you explicitly put `workers_dev = false` in your toml anywhere on its own, it's super ambiguous, and so i feel it should error. i should probably test this against today's behavior, but it's wicked inconsistent. imho.

Reply:
it's not entirely clear what `EnvConfig::zoneless(false)` is doing here - not easy for me to parse out what exactly the toml would look like and what is supposed to fail/succeed. i'm assuming `EnvConfig::zoneless(false)` means that `workers_dev = false`, in which case we might want to rename that method